packages/beacon-node/src/chain/regen/regen.ts

```diff
@@ -77,20 +84,23 @@ export class StateRegenerator implements IStateRegeneratorInternal {
   async getCheckpointState(
     cp: phase0.Checkpoint,
     opts: StateCloneOpts,
-    rCaller: RegenCaller
+    rCaller: RegenCaller,
+    shouldReload = false
```

Comment:
should this flag move in StateCloneOpts?

Reply:
`shouldReload` is designed to use within this file only, I don't want to publish `shouldReload` flag to all consumers, it should only be `true` when we get state for block processing, or `updateHeadState`. It's easier to control it within this file